docs/self-hosting/advanced/sso-provides/microsoft-entra-id.mdx

```diff
@@ -0,0 +1,75 @@
+  http(s)://<your-domain>/api/auth/callback/azure-ad
+  ```
+
+  <Callout type={'info'}>
+    - You can fill in or modify the Redirect URIs after registering, but make sure the URL you enter matches the deployed URL.
+    - Please replace "your-domain" with your own domain.
+  </Callout>
+
+  ![]()
+  
+  Click on "Register".
+
+  After successfully creating the application, click on the corresponding application to enter the application details page, and switch to the "Overview" tab to view the corresponding configuration information.
+
+  ![]()
+
+  Go to "Certificates & secrets", select the "Client secrets" tab, click on "New client secret", fill in the description, select the expiration time, and click on "Add" to create a new client secret.
+
+  ![]()
+
+  <Callout type={'info'}>
+    - Please make sure to save your client secret as this is your only chance to view it.
+  </Callout>
+
+
+  ### Add Users
+
+  Go back to the "Microsoft Entra ID" interface, enter "Users", click on "New user", fill in the user information, and click on "Create" to create a user for using LobeChat.
+
+  ### Configure Environment Variables
+
+  When deploying LobeChat, you need to configure the following environment variables:
+
+| Environment Variable     | Type       | Description                                                                                         |
+| ------------------------ | ---------- | --------------------------------------------------------------------------------------------------- |
+| `ENABLE_OAUTH_SSO`       | Required   | Enable single sign-on (SSO) for LobeChat. Set to `1` to enable single sign-on.                      |
+| `NEXTAUTH_SECRET`        | Required   | Key used to encrypt Auth.js session tokens. You can generate the key using the following command: `openssl rand -base64 32` |
+| `SSO_PROVIDERS`          | Required   | Select the single sign-on provider for LoboChat. Use `azure-ad` for Microsoft Entra ID.              | 
+| `AZURE_AD_CLIENT_ID`     | Required   | Client ID of the Microsoft Entra ID application.                                                     |
+| `AZURE_AD_CLIENT_SECRET` | Required   | Client Secret of the Microsoft Entra ID application.                                                 |
+| `AZURE_AD_TENANT_ID`     | Required   | Tenant ID of the Microsoft Entra ID application.                                                     |
+| `ACCESS_CODE`            | Required   | Add a password for accessing this service, you can set a sufficiently long random password to "disable" access code authorization. |
+| `NEXTAUTH_URL`           | Optional   | This URL is used to specify the callback address for Auth.js when performing OAuth authentication. It is only necessary to set it when the default generated redirect address is incorrect. `https://example.com/api/auth` |
+
+  You can refer to [environment variables](/en/self-hosting/environment-variable#microsoft-entra-id) for details on related variables.
```

Comment:
Is the environment-variable document missing?

Reply:
Oh, I will adding that.